overcooked_ai_py/agents/agent.py

```diff
@@ -515,3 +532,2155 @@ def ml_action(self, state):
+#     it can be queried to obtain actions for both the agents.
+#     """
+#
+#     def __init__(self, *agents):
+#         """
+#         If the pair of agents is in fact a single joint agent, set the agent
+#         index (used to order the processed observations) to 0, that is consistent
+#         with training.
+#
+#         Otherwise, set the agent indices in the same order as the agents have been passed in.
+#         """
+#         self.agents = agents
+#
+#         if len(agents) == 1:
+#             self.is_joint_agent = True
+#             self.joint_agent = agents[0]
+#             self.joint_agent.set_agent_index(0)
+#         else:
+#             self.is_joint_agent = False
+#             self.a0, self.a1 = agents
+#             self.a0.set_agent_index(0)
+#             self.a1.set_agent_index(1)
+#
+#     def set_mdp(self, mdp):
+#         for a in self.agents:
+#             a.set_mdp(mdp)
+#
+#     def joint_action(self, state):
+#         if self.is_joint_agent:
+#             joint_action = self.joint_agent.action(state)
+#             return joint_action
+#         elif type(self.a0) is CoupledPlanningAgent and type(self.a1) is CoupledPlanningAgent:
+#             # Reduce computation by half if both agents are coupled planning agents
+#             joint_action_plan = self.a0.mlp.get_low_level_action_plan(state, self.a0.heuristic, delivery_horizon=self.a0.delivery_horizon, goal_info=True)
+#             return joint_action_plan[0] if len(joint_action_plan) > 0 else (None, None)
+#         elif self.a0 is self.a1:
+#             # When using the same instance of an agent for self-play,
+#             # reset agent index at each turn to prevent overwriting it
+#             self.a0.set_agent_index(0)
+#             action_0 = self.a0.action(state)
+#             self.a1.set_agent_index(1)
+#             action_1 = self.a1.action(state)
+#             return (action_0, action_1)
+#         else:
+#             return (self.a0.action(state), self.a1.action(state))
+#
+#     def reset(self):
+#         for a in self.agents:
+#             a.reset()
+
+
+# ============================ Agents make by pk ===================================#
+
+class GreedyHumanModel_pk(Agent):
+    """
+    This is Paul's GreedyHumanModel, which is slightly different Micah's (just called GreedyHumanModel)
+
+    Agent that at each step selects a medium level action corresponding to the most intuitively high-priority thing to do
+
+    Enhancements added to v2:
+    1) Drop onion if a dish is needed instead
+    2) Drop dish if an onion is needed instead
+    3) If soup on the counter then pick it up
+    3) Added parameter "perseverance", then added mechanism for resolving crashes: each agent steps out of the way with
+    a certain probability, which depends on their perseverance and how many timesteps they've been stuck.
+
+    """
+
+    # TODO: Remove any mention of tomato!
```

Comment:
Remove this line, and logging.info calls

Reply:
The logging.info calls are there so we can see what the agents are "thinking" as they play, by setting "logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.INFO)" when viewing the agents (e.g. in view_agent.py). This is useful for debugging etc because at any timestep we can see what decision the agent is making any why.

Anyway, I guess that once I've finished making the new TOM model (and once I'm sure it's working correctly), then these logging.info calls won't be necessary, so I think it will be fine for me to remove them. Although if any later changes are made to the model then they might be useful again.

I can see that they made the code more messy, so would you advise that I remove all the logging.info calls as soon as I think the new TOM model is working okay?

Reply:
Yeah it's fine to keep the logging.info calls in your own project repo for now (given that you moved the agent, it's actually fine), was just reluctant to committing them to overcooked_ai

Reply:
Yep fair enough